elastic/kibana
# [Alerting] Using abortable es client in stack rules

Resolves https://github.com/elastic/kibana/issues/121152

## Summary

Switch to using abortable es client search for index threshold and ES query rule types. There is a functional test for using this es client in a long running rule available in `x-pack/test/alerting_api_integration/spaces_only/tests/alerting/builtin_alert_types/cancellable/rule.ts` but I don't believe I can add a functional test for these specific rule types because I would need to modify the ES query inside the executor to run long.

## To Verify

1. Create an index threshold rule and and ES query rule and verify that they run correctly with no issues.
2. Modify the ES query rule type to set a short timeout and a long query delay

```
--- a/x-pack/plugins/stack_alerts/server/alert_types/es_query/alert_type.ts
+++ b/x-pack/plugins/stack_alerts/server/alert_types/es_query/alert_type.ts
@@ -148,6 +148,7 @@ export function getAlertType(logger: Logger): RuleType<
     isExportable: true,
     executor,
     producer: STACK_ALERTS_FEATURE_ID,
+    ruleTaskTimeout: '10s',
   };

   async function executor(
@@ -220,6 +221,13 @@ export function getAlertType(logger: Logger): RuleType<
       searchAfterSortId: undefined,
       timeField: params.timeField,
       track_total_hits: true,
+      aggs: {
+        delay: {
+          shard_delay: {
+            value: '30s',
+          },
+        },
+      },
     });

     logger.debug(`alert ${ES_QUERY_ID}:${alertId} "${name}" query - ${JSON.stringify(query)}`);
```

3. Verify that executions for this rule times out and a timeout error is thrown
4. Modify the index threshold rule type to set a short timeout and a long query delay

```
--- a/x-pack/plugins/stack_alerts/server/alert_types/index_threshold/alert_type.ts
+++ b/x-pack/plugins/stack_alerts/server/alert_types/index_threshold/alert_type.ts
@@ -128,6 +128,7 @@ export function getAlertType(
     isExportable: true,
     executor,
     producer: STACK_ALERTS_FEATURE_ID,
+    ruleTaskTimeout: '10s',
   };

   async function executor(

--- a/x-pack/plugins/triggers_actions_ui/server/data/lib/time_series_query.ts
+++ b/x-pack/plugins/triggers_actions_ui/server/data/lib/time_series_query.ts
@@ -95,6 +95,11 @@ export async function timeSeriesQuery(
         ranges: dateRangeInfo.dateRanges,
       },
     },
+    delay: {
+      shard_delay: {
+        value: '30s',
+      },
+    },
   };

   // if not count, add a sorted value agg
```

5. Verify that executions for this rule times out and a timeout error is thrown

### Checklist

Delete any items that are not applicable to this PR.
